### PR TITLE
added cart-management

### PR DESCRIPTION
**Pull request by Manpreet-hub**
### Features:- 
- Increase and decrease the count of the item.
- Add to cart and Remove from cart.

### CODE REVIEW

File: src/components/Cart/CartItem.jsx [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+        <>
+
+            {
+                    cart.map((item) => {
+                    return (
+                        <div className="horizontal-card  product-card">
+                            <img className="card-img product-img" src={item.img} alt="product-img" />
+                            <div className="card-body">
+                                <h4 className="card-title"> {item.title}</h4>
+                                {/* <p className="card-text">Peter England Men's Notch Lapel Slim Blazer.</p> */}
```
Comment (abhijeetekad) [inline src/components/Cart/CartItem.jsx]: Remove this commented code.

File: src/contexts/cart-context.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+
+const CartContext=createContext();
+const useCart=()=>useContext(CartContext);
+
+const CartProvider=({children})=>{
+    
+    const [cartState,cartDispatch]=useReducer(cartReducer,cartInitialState);
+    console.log(cartState)
+
```
Comment (abhijeetekad) [inline src/contexts/cart-context.js]: Remove console.log

File: src/styles/cart.css [hunk 3]
```diff
@@ -0,0 +1,61 @@
+.cart-container{
+    padding: 5rem 0;
+    height: 100vh;
+    /* background-color: var(--lynx-white-color); */
```
Comment (abhijeetekad) [inline src/styles/cart.css]: +1

**Review by abhijeetekad — APPROVED** [review]
The code looks clean and neat.

File: src/components/Cart/CartSummary.jsx [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+import { useCart } from '../../contexts/cart-context';
+
+const CartSummary=()=> {
+    const {cartState:{cart}}=useCart();
+    const totalAmt=cart.reduce((acc,cur)=> Number(cur.price) * Number(cur.qty) + acc,0);
+    const totalItem=cart.reduce((acc,cur)=> Number(cur.qty) + acc,0);
+    const deliveryCharges=100;
+    const finalAmt=totalAmt+deliveryCharges;
+
+    console.log(totalAmt);
```
Comment (prathmeshjagtap) [inline src/components/Cart/CartSummary.jsx]: you can remove console.log

File: src/components/ProductCard.jsx [hunk 5]
```diff
@@ -9,7 +12,21 @@ const ProductCard=({product})=>{
                     <h4 className="card-title"> {title}</h4>
                     <p>₹{price}</p>
                     <p >Rating : {rating}</p>
-                    <button className="btn-default btn-primary">Add to Cart</button>
+
+                    {
+                        cart.some(p=>p.id === product.id) ?(
```
Comment (prathmeshjagtap) [inline src/components/ProductCard.jsx]: you can give some descriptive names instead  of p

File: src/reducers/cart-reducer.js [hunk 6]
```diff
@@ -0,0 +1,32 @@
+const cartInitialState={
+    cart:[],
+}
```
Comment (prathmeshjagtap) [inline src/reducers/cart-reducer.js:3]: you can keep this initial state in the context itself

File: src/styles/cart.css [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+
+.cart-btn{
+    width: 95%;
+}
+
+.billing-details{
+    width: 18%;
+    padding: 2rem;
+    background-color: var(--white-color);
+    /* display: flex; */
```
Comment (prathmeshjagtap) [inline src/styles/cart.css]: remove if not needed

**Review by prathmeshjagtap — CHANGES_REQUESTED** [review]

**Review by prathmeshjagtap — APPROVED** [review]
Work on the changes suggested rest everything is good

File: src/styles/cart.css (hunk 3 above)
Comment (Manpreet-hub) [reply src/styles/cart.css]: Done [346210e](https://github.com/Manpreet-hub/ecommerce-react/pull/4/commits/346210efcfbdd93bcf04432da991ab1028dd18a2)

File: src/components/Cart/CartSummary.jsx (hunk 4 above)
Comment (Manpreet-hub) [reply src/components/Cart/CartSummary.jsx]: Done [346210e](https://github.com/Manpreet-hub/ecommerce-react/pull/4/commits/346210efcfbdd93bcf04432da991ab1028dd18a2)

File: src/components/ProductCard.jsx (hunk 5 above)
Comment (Manpreet-hub) [reply src/components/ProductCard.jsx]: done [19e27c0](https://github.com/Manpreet-hub/ecommerce-react/pull/4/commits/19e27c012f8563009429d4c1ee8d97b16b63ecc9)

File: src/styles/cart.css (hunk 7 above)
Comment (Manpreet-hub) [reply src/styles/cart.css]: Done [346210e](https://github.com/Manpreet-hub/ecommerce-react/pull/4/commits/346210efcfbdd93bcf04432da991ab1028dd18a2)

File: src/reducers/cart-reducer.js (hunk 6 above)
Comment (Manpreet-hub) [reply src/reducers/cart-reducer.js:3]: Will do it while refactoring the code.

**Pull request merged**